tests/integration/cmor/_fixes/icon/test_icon.py

```diff
@@ -0,0 +1,445 @@
+"""Tests for the ICON on-the-fly CMORizer."""
+from functools import lru_cache
+
+import iris
+import numpy as np
+import pytest
+from cf_units import Unit
+
+from esmvalcore._config import get_extra_facets
+from esmvalcore.cmor._fixes.icon.icon import AllVars, Siconca
+from esmvalcore.cmor.fix import Fix
+from esmvalcore.cmor.table import get_var_info
+
+
+@lru_cache
+@pytest.fixture
+def cubes_2d(test_data_path):
+    """2D sample cubes."""
+    nc_path = test_data_path / 'icon_2d.nc'
+    return iris.load(str(nc_path))
+
+
+@lru_cache
+@pytest.fixture
+def cubes_3d(test_data_path):
+    """3D sample cubes."""
+    nc_path = test_data_path / 'icon_3d.nc'
+    return iris.load(str(nc_path))
+
+
+@lru_cache
```

Comment:
Please don't use `lru_cache` for caching fixtures, because there is a suspicion that this causes segmenation faults in the tests. You can set scope="module" if you want to re-use the fixture for the entire module, see https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session.